### PR TITLE
Add button mission type list

### LINKED ISSUE #306: Add a delete button in the mission types list
=> enlever le bouton "supprimer" de la page d'édition des types de mission et le placer dans la liste, comme pour les véhicules
=> supprimer la protection csrf

### PR DESCRIPTION
**Pull request by olivmai**
- add delete button in mission_type list
- edit mission_type index template according to style applied on vehicles list
- remove delete_form (delete from edit template and delete form template itself) for mission_type as it is not needed anymore

I did this consistent to what was already done in vehicles list

Closes #306

### CODE REVIEW

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+class MissionTypeDeleteController extends AbstractController
+{
+    public function __invoke(EntityManagerInterface $entityManager, MissionType $missionType): RedirectResponse
+    {
+        $organization = $this->getUser();
+        if (!$organization instanceof Organization || false === $organization->isParent()) {
+            throw new AccessDeniedException();
+        }
+
+        $entityManager->beginTransaction();
```
Comment (mRoca) [inline src/Controller/Organization/MissionType/MissionTypeDeleteController.php:27]: this is already done in the flush()

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+/**
+ * @Route("/delete/{id}", name="app_organization_mission_type_delete", methods={"GET"})
+ */
+class MissionTypeDeleteController extends AbstractController
+{
+    public function __invoke(EntityManagerInterface $entityManager, MissionType $missionType): RedirectResponse
```
Comment (mRoca) [inline src/Controller/Organization/MissionType/MissionTypeDeleteController.php:20]: please use the @ security tag to protect the controller

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        if (!$organization instanceof Organization || false === $organization->isParent()) {
+            throw new AccessDeniedException();
+        }
+
+        $entityManager->beginTransaction();
+        $entityManager->remove($missionType);
+        $entityManager->flush();
+        $entityManager->commit();
+
+        $this->addFlash('success', 'Le type de mission a été supprimé avec succès.');
```
Comment (mRoca) [inline src/Controller/Organization/MissionType/MissionTypeDeleteController.php:32]: can you use translations ?

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * @IsGranted("ROLE_PARENT_ORGANIZATION")
+ */
+class MissionTypeDeleteController extends AbstractController
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private TranslatorInterface $translator;
+
+    public function __construct(TranslatorInterface $translator)
```
Comment (mRoca) [inline src/Controller/Organization/MissionType/MissionTypeDeleteController.php:30]: you should inject `EntityManagerInterface $entityManager` here also

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * @Route("/delete/{id}", name="app_organization_mission_type_delete", methods={"GET"})
+ * @Security("missionType.organization == user")
```
Comment (mRoca) [inline src/Controller/Organization/MissionType/MissionTypeDeleteController.php:20]: are you sure this security tag shouldn't be on the __invoke method ?

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+ * @IsGranted("ROLE_PARENT_ORGANIZATION")
+ */
+class MissionTypeDeleteController extends AbstractController
+{
+    /**
+     * @var TranslatorInterface
```
Comment (mRoca) [inline src/Controller/Organization/MissionType/MissionTypeDeleteController.php:26]: to remove, the typehint is enough

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function __invoke(EntityManagerInterface $entityManager, MissionType $missionType): RedirectResponse
+    {
+        $organization = $this->getUser();
+        if (!$organization instanceof Organization || false === $organization->isParent()) {
```
Comment (mRoca) [inline src/Controller/Organization/MissionType/MissionTypeDeleteController.php:38]: to remove, you can use `@var Organization $organization` instead

File: templates/organization/mission_type/index.html.twig [hunk 8]
```diff
@@ -2,31 +2,63 @@
...
+{% endblock %}
+
 {% block body %}
     <h1>Types de missions</h1>
 
-    <table class="table mt-5">
-        <thead>
+    <div class="row">
+        <div class="col-12 text-right mb-2">
+            <a class="btn btn-success" href="{{ path('app_organization_mission_type_new') }}" role="button">Ajouter un nouveau type de mission</a>
```
Comment (mRoca) [inline templates/organization/mission_type/index.html.twig:15]: translation

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php (hunk 1 above)
Comment (olivmai) [reply src/Controller/Organization/MissionType/MissionTypeDeleteController.php:42]: Indeed, I took it from vehicles controller but was not sure it is really usefull...

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php (hunk 2 above)
Comment (olivmai) [reply src/Controller/Organization/MissionType/MissionTypeDeleteController.php:35]: yes, update coming soon ;)

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php (hunk 3 above)
Comment (olivmai) [reply src/Controller/Organization/MissionType/MissionTypeDeleteController.php]: yep, done

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php (hunk 4 above)
Comment (olivmai) [reply src/Controller/Organization/MissionType/MissionTypeDeleteController.php:30]: Indeed, well seen :)

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php (hunk 5 above)
Comment (olivmai) [reply src/Controller/Organization/MissionType/MissionTypeDeleteController.php:20]: I don't know what is best practice but it seems to work like that and as it is a dedicated controller (for MissionType removal) with only one __invoke method, i suppose it's ok to do like that ?

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php (hunk 6 above)
Comment (olivmai) [reply src/Controller/Organization/MissionType/MissionTypeDeleteController.php:26]: ok

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php (hunk 7 above)
Comment (olivmai) [reply src/Controller/Organization/MissionType/MissionTypeDeleteController.php:38]: ok

File: templates/organization/mission_type/index.html.twig (hunk 8 above)
Comment (olivmai) [reply templates/organization/mission_type/index.html.twig:15]: Yes, i will check in all modified templates if there are another missing translations

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php (hunk 1 above)
Comment (olivmai) [reply src/Controller/Organization/MissionType/MissionTypeDeleteController.php:42]: @mRoca  what do you think about the 
```php 
$entityManager->commit();
```
statement in the same code block ? I'm wondering if it is really usefull too...

Comment (olivmai) [reply src/Controller/Organization/MissionType/MissionTypeDeleteController.php]: I have my answer. commit() is not needed anymore as we removed the openTransaction() call

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php (hunk 5 above)
Comment (mRoca) [reply src/Controller/Organization/MissionType/MissionTypeDeleteController.php:20]: If you're sure it's working like this, it's okay for me :-)

File: src/Controller/Organization/MissionType/MissionTypeDeleteController.php [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    public function __construct(TranslatorInterface $translator, EntityManagerInterface $entityManager)
+    {
+        $this->translator = $translator;
+        $this->entityManager = $entityManager;
+    }
+
+    public function __invoke(MissionType $missionType): RedirectResponse
+    {
+        /** @var Organization $this->getUser() */
+        if (!$this->getUser() instanceof Organization || false === $this->getUser()->isParent()) {
```
Comment (mRoca) [inline src/Controller/Organization/MissionType/MissionTypeDeleteController.php:37]: to remove, the test is already done in `@IsGranted("ROLE_PARENT_ORGANIZATION")`

**Pull request merged**